### PR TITLE
Add --overwrite option to rnp

### PR DESCRIPTION
**Pull request by ni4**
Before now rnp was overwriting output file by default, and without any option to change this behavior.
So this PR:
- adds `--overwrite` option, which will allow to overwrite output by default
- fixes weird double fd opening during file signing (should be typo)
- updates handling of output stream creation with asking user whether to overwrite file/new filename

### CODE REVIEW

**Comment by flowher** [comment]
"Before now rnp was overwriting output file by default", is it really true? Or do you mean that it wasn't overwriting?

File: src/rnp/rnpcfg.c [hunk 1]
```diff
@@ -306,17 +306,12 @@ bool
...
-        if ((strcmp(val, "true") == 0) || (strcmp(val, "True") == 0)) {
-            return true;
-        } else if (atoi(val) > 0) {
-            return true;
-        } else
-            return false;
-    } else {
-        return false;
+    
+    if (val && ((strcmp(val, "true") == 0) || (strcmp(val, "True") == 0) || (atoi(val) > 0))) {
```
Comment (flowher) [inline src/rnp/rnpcfg.c]: better to use ``tolower()`` and then compare to "true"

Comment (flowher) [inline src/rnp/rnpcfg.c]: easier to read if "return (val && ((strcmp(val, "true") == 0) || (strcmp(val, "True") == 0) || (atoi(val) > 0))) "

File: src/lib/misc.c [hunk 2]
```diff
@@ -1134,3 +1134,15 @@ rnp_strlwr(char *s)
...
+rnp_strip_eol(char *s)
+{
+    size_t len = strlen(s);
+
+    while (len > 0 && (s[len - 1] == '\n' || s[len - 1] == '\r')) {
+        s[--len] = '\0';
+    }
+
+    return s;
+}
```
Comment (flowher) [inline src/lib/misc.c]: \n

File: src/lib/rnp.c [hunk 3]
```diff
@@ -1200,11 +1200,116 @@ rnp_decrypt_file(rnp_ctx_t *ctx, const char *f, const char *out)
...
+ *  @param dst Allocated dest structure to put result in
+ *  @param out Output filename/path as provided by user. For NULL or '-' stdout will be used.
+ *  @return true on success or false otherwise
+ **/
+
+static bool
+rnp_initialize_output(rnp_ctx_t *ctx, pgp_dest_t *dst, const char *out)
+{
+    char newname[PATH_MAX];
+    bool is_stdout = out == NULL || out[0] == '\0' || strcmp(out, "-") == 0;
```
Comment (flowher) [inline src/lib/rnp.c]: can you put () around those conditions. It hard to read this way

**Comment by flowher** [comment]
LGTM, some minor comments added

File: src/rnp/rnpcfg.c (hunk 1 above)
Comment (ronaldtse) [reply src/rnp/rnpcfg.c]: Agree.

Comment (ronaldtse) [reply src/rnp/rnpcfg.c]: Agree.

Comment (ni4) [reply src/rnp/rnpcfg.c]: Used strcasecmp - otherwise would need to check val, allocate buffer for lowercase copy, deallocate, etc, etc.

Comment (ni4) [reply src/rnp/rnpcfg.c]: Done.

File: src/lib/misc.c (hunk 2 above)
Comment (ni4) [reply src/lib/misc.c]: Ok

File: src/lib/rnp.c (hunk 3 above)
Comment (ni4) [reply src/lib/rnp.c]: Ok.

**Comment by ni4** [comment]
@flowher @ronaldtse Thanks for comments, fixed.

> "Before now rnp was overwriting output file by default", is it really true? Or do you mean that it wasn't overwriting?

Yeah, it was. overwriting without the prompt. It has hardcoded setting overwrite to 1, so I moved the same to rnpcfg that time.

**Comment by ni4** [comment]
@flowher Any other comments, or this may be merged?

File: src/lib/misc.c [hunk 4]
```diff
@@ -1134,3 +1134,15 @@ rnp_strlwr(char *s)
     }
     return s;
 }
+
+char *
+rnp_strip_eol(char *s)
+{
+    size_t len = strlen(s);
+
+    while (len > 0 && (s[len - 1] == '\n' || s[len - 1] == '\r')) {
```
Comment (flowher) [inline src/lib/misc.c]: () here also please

File: include/rnp/rnp_sdk.h [hunk 5]
```diff
@@ -71,4 +71,6 @@ bool rnp_hex_decode(const char *hex, uint8_t *buf, size_t buf_len);
 
 char *rnp_strlwr(char *s);
 
+char *rnp_strip_eol(char *s);
```
Comment (flowher) [inline include/rnp/rnp_sdk.h]: Please add comment what this function really does

**Comment by flowher** [comment]
Two additional comments. Go ahead and merge after they're done

File: src/lib/misc.c (hunk 4 above)
Comment (ni4) [reply src/lib/misc.c]: Done

File: include/rnp/rnp_sdk.h (hunk 5 above)
Comment (ni4) [reply include/rnp/rnp_sdk.h]: Done

**Pull request merged**